src/overlays/actors/ovl_Bg_Mori_Hashigo/z_bg_mori_hashigo.c

```diff
@@ -20,39 +40,255 @@ const ActorInit Bg_Mori_Hashigo_InitVars = {
+        // Bank danger!
+        osSyncPrintf("Error : バンク危険！(arg_data 0x%04x)(%s %d)\n", this->dyna.actor.params,
+                     "../z_bg_mori_hashigo.c", 312);
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        BgMoriHashigo_SetupObjectCheck(this);
+        // (Forest Temple Ladder and its clasp)
+        osSyncPrintf("(森の神殿 梯子とその留め金)(arg_data 0x%04x)\n", this->dyna.actor.params);
+    }
+}
+
+void BgMoriHashigo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriHashigo* this = THIS;
+
+    if (this->dyna.actor.params == 0) {
+        DynaPolyInfo_Free(globalCtx2, &globalCtx2->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+    if (this->dyna.actor.params == (s16)0xFFFF) {
+        Collider_DestroyJntSph(globalCtx, &this->collider);
+    }
+}
+
+void BgMoriHashigo_SetupObjectCheck(BgMoriHashigo* this) {
+    this->actionFunc = BgMoriHashigo_ObjectCheck;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/func_808A2698.s")
+void BgMoriHashigo_ObjectCheck(BgMoriHashigo* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex)) {
+        if (this->dyna.actor.params == (s16)0xFFFF) {
+            BgMoriHashigo_SetupClasp(this);
+        } else if (this->dyna.actor.params == 0) {
+            BgMoriHashigo_SetupLadderWait(this);
+        }
+        this->dyna.actor.draw = BgMoriHashigo_Draw;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/func_808A2770.s")
+void BgMoriHashigo_SetupClasp(BgMoriHashigo* this) {
+    this->actionFunc = BgMoriHashigo_Clasp;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/func_808A27F8.s")
+void BgMoriHashigo_Clasp(BgMoriHashigo* this, GlobalContext* globalCtx) {
+    if (this->hitTimer <= 0) {
+        if (this->collider.base.acFlags & 2) {
+            this->collider.base.acFlags &= ~2;
+            this->hitTimer = 10;
+        } else {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/BgMoriHashigo_Init.s")
+void BgMoriHashigo_SetupLadderWait(BgMoriHashigo* this) {
+    this->actionFunc = BgMoriHashigo_LadderWait;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/BgMoriHashigo_Destroy.s")
+void BgMoriHashigo_LadderWait(BgMoriHashigo* this, GlobalContext* globalCtx) {
+    BgMoriHashigo* clasp = (BgMoriHashigo*)this->dyna.actor.parent;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/func_808A2978.s")
+    if (clasp->hitTimer > 0) {
+        BgMoriHashigo_LadderFall(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hashigo/func_808A2988.s")
+void BgMoriHashigo_LadderFall(BgMoriHashigo* this) {
+    this->bounceCounter = 0;
+    this->actionFunc = BgMoriHashigo_LadderBounce;
```

Comment:
```suggestion
void BgMoriHashigo_SetupLadderFreed(BgMoriHashigo* this) {
    this->bounceCounter = 0;
    this->actionFunc = BgMoriHashigo_LadderFreed;
```
I don't really like `Freed` but I think keeping the SetupActionName / ActionName pattern is important

`Unclasped`?

Reply:
I think this is fine as SetupFall and Fall